diff --git a/CMakeLists.txt b/CMakeLists.txt
index fbc2fde..419ce4d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -204,7 +204,7 @@ ELSE()
 ENDIF()
 
 CONFIGURE_FILE(
-    "${PROJECT_SOURCE_DIR}/config.h.cmake"
+    "${PROJECT_SOURCE_DIR}/config.h.in"
     "${PROJECT_BINARY_DIR}/config.h")
 # Necessary for the compiler to find "config.h":
 INCLUDE_DIRECTORIES("${PROJECT_BINARY_DIR}")
diff --git a/configure b/configure
index 182fba2..3448c24 100755
--- a/configure
+++ b/configure
@@ -16367,8 +16367,8 @@ ac_res=$ac_cv_search_XML_StopParser
 if test "$ac_res" != no; then :
   test "$ac_res" = "none required" || LIBS="$ac_res $LIBS"
 
-else
-  as_fn_error $? "cannot find EXPAT function XML_StopParser" "$LINENO" 5
+# else
+#   as_fn_error $? "cannot find EXPAT function XML_StopParser" "$LINENO" 5
 fi
 
 
diff --git a/configure.ac b/configure.ac
index c611a11..d75b177 100644
--- a/configure.ac
+++ b/configure.ac
@@ -102,8 +102,8 @@ AC_SEARCH_LIBS([dirname], [gen], ,
     AC_MSG_ERROR([cannot find function dirname]))
 AC_SEARCH_LIBS([log10], [m], , 
     AC_MSG_ERROR([cannot find function log10]))
-AC_SEARCH_LIBS([XML_StopParser], [expat], , 
-    AC_MSG_ERROR([cannot find EXPAT function XML_StopParser]))
+# AC_SEARCH_LIBS([XML_StopParser], [expat], , 
+#     AC_MSG_ERROR([cannot find EXPAT function XML_StopParser]))
 
 AC_CHECK_LIB([dl], [dlopen])
 