diff --git a/cf/units.py b/cf/units.py
index 4948680..742db1a 100644
--- a/cf/units.py
+++ b/cf/units.py
@@ -1,12 +1,7 @@
-from ctypes.util import find_library
-
+import cfunits
 from cfunits import Units as cfUnits
 
-_libpath = find_library("udunits2")
-if _libpath is None:
-    raise FileNotFoundError(
-        "cf requires UNIDATA UDUNITS-2. Can't find the 'udunits2' library."
-    )
+_libpath = _libpath = cfunits.units._udunits._name
 
 
 class Units:
