diff --git a/src/eckit/CMakeLists.txt b/src/eckit/CMakeLists.txt
index 6a0766e..654b046 100644
--- a/src/eckit/CMakeLists.txt
+++ b/src/eckit/CMakeLists.txt
@@ -810,7 +810,7 @@ elseif( EC_OS_NAME STREQUAL "freebsd" )
 elseif( EC_OS_NAME STREQUAL "macosx" )
     list( APPEND eckit_system_srcs system/SystemInfoMacOSX.h system/SystemInfoMacOSX.cc )
 else()
-    ecbuild_warn("eckit does not support SystemInfo for OS '${EC_OS_NAME}'")
+    list( APPEND eckit_system_srcs system/SystemInfoEmscripten.h system/SystemInfoEmscripten.cc )
 endif()
 
 list( APPEND eckit_bases_srcs
diff --git a/src/eckit/system/SystemInfo.cc b/src/eckit/system/SystemInfo.cc
index 96ddb01..e9473d9 100644
--- a/src/eckit/system/SystemInfo.cc
+++ b/src/eckit/system/SystemInfo.cc
@@ -41,6 +41,10 @@
 #include "eckit/system/SystemInfoFreeBSD.h"
 #endif
 
+#if defined(__EMSCRIPTEN__)
+#include "eckit/system/SystemInfoEmscripten.h"
+#endif
+
 namespace eckit::system {
 
 //----------------------------------------------------------------------------------------------------------------------
@@ -66,6 +70,10 @@ SystemInfo* makeSystemInfo(const std::string& system) {
     }
 #endif
 
+#if defined(__EMSCRIPTEN__)
+return new SystemInfoEmscripten();
+#endif
+
     NOTIMP;
 }
 
diff --git a/src/eckit/system/SystemInfoEmscripten.cc b/src/eckit/system/SystemInfoEmscripten.cc
new file mode 100644
index 0000000..e94a2f4
--- /dev/null
+++ b/src/eckit/system/SystemInfoEmscripten.cc
@@ -0,0 +1,49 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+
+/// @author Juniper Tyree
+/// @date   April 2025
+
+#include "eckit/system/SystemInfoEmscripten.h"
+
+#include "eckit/exception/Exceptions.h"
+#include "eckit/filesystem/LocalPathName.h"
+#include "eckit/io/Buffer.h"
+#include "eckit/system/MemoryInfo.h"
+
+namespace eckit {
+namespace system {
+
+//----------------------------------------------------------------------------------------------------------------------
+
+SystemInfoEmscripten::~SystemInfoEmscripten() {}
+
+std::string SystemInfoEmscripten::userName() const {
+    return std::string("pyodide");
+}
+
+LocalPathName SystemInfoEmscripten::executablePath() const {
+    return LocalPathName(std::string("/lib/python314.zip")).realName();
+}
+
+MemoryInfo SystemInfoEmscripten::memoryUsage() const {
+    MemoryInfo mem;
+    return mem;
+}
+
+std::string SystemInfoEmscripten::dynamicLibraryName(const std::string& name) const {
+    return std::string("lib") + name + ".so";
+}
+
+
+//----------------------------------------------------------------------------------------------------------------------
+
+}  // namespace system
+}  // namespace eckit
diff --git a/src/eckit/system/SystemInfoEmscripten.h b/src/eckit/system/SystemInfoEmscripten.h
new file mode 100644
index 0000000..4ed2e95
--- /dev/null
+++ b/src/eckit/system/SystemInfoEmscripten.h
@@ -0,0 +1,46 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+
+/// @author Juniper Tyree
+/// @date   April 2025
+
+#ifndef eckit_system_SystemInfoEmscripten_H
+#define eckit_system_SystemInfoEmscripten_H
+
+#include "eckit/system/SystemInfo.h"
+
+namespace eckit {
+namespace system {
+
+//----------------------------------------------------------------------------------------------------------------------
+
+class SystemInfoEmscripten : public SystemInfo {
+
+public:  // methods
+    ~SystemInfoEmscripten() override;
+
+    virtual std::string userName() const;
+
+    virtual eckit::LocalPathName executablePath() const;
+
+    virtual MemoryInfo memoryUsage() const;
+
+    virtual std::string dynamicLibraryName(const std::string& name) const;
+
+protected:  // methods
+private:    // members
+};
+
+//----------------------------------------------------------------------------------------------------------------------
+
+}  // namespace system
+}  // namespace eckit
+
+#endif
