diff --git a/cfdm/__init__.py b/cfdm/__init__.py
index c74ad69..b6808ed 100644
--- a/cfdm/__init__.py
+++ b/cfdm/__init__.py
@@ -62,6 +62,19 @@ _requires = core._requires + (
 
 _error0 = f"cfdm requires the modules {', '.join(_requires)}. "
 
+# Check the version of h5py
+try:
+    import h5py
+except ImportError as error1:
+    raise ImportError(_error0 + str(error1))
+else:
+    _minimum_vn = "3.12.0"
+    if Version(h5py.__version__) < Version(_minimum_vn):
+        raise ValueError(
+            f"Bad h5py version: cfdm requires h5py>={_minimum_vn}. "
+            f"Got {h5py.__version__} at {h5py.__file__}"
+        )
+
 # Check the version of cftime
 try:
     import cftime
@@ -101,19 +114,6 @@ else:
             f"Got {h5netcdf.__version__} at {h5netcdf.__file__}"
         )
 
-# Check the version of h5py
-try:
-    import h5py
-except ImportError as error1:
-    raise ImportError(_error0 + str(error1))
-else:
-    _minimum_vn = "3.12.0"
-    if Version(h5py.__version__) < Version(_minimum_vn):
-        raise ValueError(
-            f"Bad h5py version: cfdm requires h5py>={_minimum_vn}. "
-            f"Got {h5py.__version__} at {h5py.__file__}"
-        )
-
 # Check the version of zarr
 try:
     import zarr
