diff --git a/scipy/integrate/meson.build b/scipy/integrate/meson.build
index 23a715d..cfaa927 100644
--- a/scipy/integrate/meson.build
+++ b/scipy/integrate/meson.build
@@ -154,7 +154,7 @@ py3.extension_module('_dop',
   f2py_gen.process('dop.pyf'),
   link_with: [dop_lib],
   c_args: [Wno_unused_variable],
-  dependencies: [fortranobject_dep],
+  dependencies: [lapack, fortranobject_dep],
   link_args: version_link_args,
   install: true,
   link_language: 'fortran',
diff --git a/scipy/optimize/meson.build b/scipy/optimize/meson.build
index d6c20d3..c4ca44b 100644
--- a/scipy/optimize/meson.build
+++ b/scipy/optimize/meson.build
@@ -101,7 +101,7 @@ py3.extension_module('_lbfgsb',
   ],
   fortran_args: fortran_ignore_warnings,
   link_args: version_link_args,
-  dependencies: [lapack_dep, blas_dep, fortranobject_dep],
+  dependencies: [lapack, lapack_dep, blas_dep, fortranobject_dep],
   override_options: ['b_lto=false'],
   install: true,
   link_language: 'fortran',
@@ -125,7 +125,7 @@ py3.extension_module('_cobyla',
   c_args: [Wno_unused_variable],
   fortran_args: fortran_ignore_warnings,
   link_args: version_link_args,
-  dependencies: [fortranobject_dep],
+  dependencies: [lapack, fortranobject_dep],
   install: true,
   link_language: 'fortran',
   subdir: 'scipy/optimize'
@@ -135,7 +135,7 @@ py3.extension_module('_minpack2',
   [f2py_gen.process('minpack2/minpack2.pyf'), 'minpack2/dcsrch.f', 'minpack2/dcstep.f'],
   fortran_args: fortran_ignore_warnings,
   link_args: version_link_args,
-  dependencies: [fortranobject_dep],
+  dependencies: [lapack, fortranobject_dep],
   override_options: ['b_lto=false'],
   install: true,
   link_language: 'fortran',
@@ -146,7 +146,7 @@ py3.extension_module('_slsqp',
   [f2py_gen.process('slsqp/slsqp.pyf'), 'slsqp/slsqp_optmz.f'],
   fortran_args: fortran_ignore_warnings,
   link_args: version_link_args,
-  dependencies: [fortranobject_dep],
+  dependencies: [lapack, fortranobject_dep],
   install: true,
   link_language: 'fortran',
   subdir: 'scipy/optimize'
diff --git a/scipy/stats/meson.build b/scipy/stats/meson.build
index bb43e3b..358279a 100644
--- a/scipy/stats/meson.build
+++ b/scipy/stats/meson.build
@@ -36,7 +36,7 @@ py3.extension_module('_mvn',
   # Wno-surprising is to suppress a pointless warning with GCC 10-12
   # (see GCC bug 98411: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=98411)
   fortran_args: [fortran_ignore_warnings, _fflag_Wno_surprising],
-  dependencies: [fortranobject_dep],
+  dependencies: [lapack, fortranobject_dep],
   link_args: version_link_args,
   install: true,
   link_language: 'fortran',
