diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2b0d974..0080f3f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,10 +74,10 @@ ecbuild_debug("ECCODES_BIG_ENDIAN=${ECCODES_BIG_ENDIAN}")
 ecbuild_debug("ECCODES_LITTLE_ENDIAN=${ECCODES_LITTLE_ENDIAN}")
 ecbuild_info("Operating system=${CMAKE_SYSTEM} (${EC_OS_BITS} bits)")
 
-# Only support 64 bit operating systems
-if( NOT EC_OS_BITS EQUAL "64" )
-    ecbuild_critical( "Operating system ${CMAKE_SYSTEM} (${EC_OS_BITS} bits) -- ecCodes only supports 64 bit platforms" )
-endif()
+# # Only support 64 bit operating systems
+# if( NOT EC_OS_BITS EQUAL "64" )
+#     ecbuild_critical( "Operating system ${CMAKE_SYSTEM} (${EC_OS_BITS} bits) -- ecCodes only supports 64 bit platforms" )
+# endif()
 
 ###############################################################################
 # some variables/options of this project
@@ -463,8 +463,6 @@ if( HAVE_BUILD_TOOLS )
 endif()
 add_subdirectory( fortran )
 
-add_subdirectory( tests )
-add_subdirectory( examples )
 add_subdirectory( data )
 add_subdirectory( samples )
 add_subdirectory( ifs_samples ) # must come after samples
diff --git a/src/eccodes/accessor/DataG1SecondOrderGeneralExtendedPacking.cc b/src/eccodes/accessor/DataG1SecondOrderGeneralExtendedPacking.cc
index 0954ff3..3c33e47 100644
--- a/src/eccodes/accessor/DataG1SecondOrderGeneralExtendedPacking.cc
+++ b/src/eccodes/accessor/DataG1SecondOrderGeneralExtendedPacking.cc
@@ -20,7 +20,7 @@ namespace eccodes::accessor
 #define MAX_NUMBER_OF_GROUPS 65534
 #define EFDEBUG              0
 
-static const unsigned long nbits[64] = {
+static const unsigned long nbits[32] = {
     0x1, 0x2, 0x4, 0x8,
     0x10, 0x20, 0x40, 0x80,
     0x100, 0x200, 0x400, 0x800,
@@ -28,15 +28,7 @@ static const unsigned long nbits[64] = {
     0x10000, 0x20000, 0x40000, 0x80000,
     0x100000, 0x200000, 0x400000, 0x800000,
     0x1000000, 0x2000000, 0x4000000, 0x8000000,
-    0x10000000, 0x20000000, 0x40000000, 0x80000000,
-    0x100000000, 0x200000000, 0x400000000, 0x800000000,
-    0x1000000000, 0x2000000000, 0x4000000000, 0x8000000000,
-    0x10000000000, 0x20000000000, 0x40000000000, 0x80000000000,
-    0x100000000000, 0x200000000000, 0x400000000000, 0x800000000000,
-    0x1000000000000, 0x2000000000000, 0x4000000000000, 0x8000000000000,
-    0x10000000000000, 0x20000000000000, 0x40000000000000, 0x80000000000000,
-    0x100000000000000, 0x200000000000000, 0x400000000000000, 0x800000000000000,
-    0x1000000000000000, 0x2000000000000000, 0x4000000000000000, 0x8000000000000000
+    0x10000000, 0x20000000, 0x40000000, 0x80000000
 };
 
 long number_of_bits(grib_handle* h, unsigned long x)
diff --git a/src/eccodes/accessor/SecondOrderBitsPerValue.cc b/src/eccodes/accessor/SecondOrderBitsPerValue.cc
index 42a19e2..0dd4fa9 100644
--- a/src/eccodes/accessor/SecondOrderBitsPerValue.cc
+++ b/src/eccodes/accessor/SecondOrderBitsPerValue.cc
@@ -17,7 +17,7 @@ eccodes::Accessor* grib_accessor_second_order_bits_per_value = &_grib_accessor_s
 namespace eccodes::accessor
 {
 
-static const size_t nbits[64] = {
+static const size_t nbits[32] = {
     0x1, 0x2, 0x4, 0x8,
     0x10, 0x20, 0x40, 0x80,
     0x100, 0x200, 0x400, 0x800,
@@ -25,15 +25,7 @@ static const size_t nbits[64] = {
     0x10000, 0x20000, 0x40000, 0x80000,
     0x100000, 0x200000, 0x400000, 0x800000,
     0x1000000, 0x2000000, 0x4000000, 0x8000000,
-    0x10000000, 0x20000000, 0x40000000, 0x80000000,
-    0x100000000, 0x200000000, 0x400000000, 0x800000000,
-    0x1000000000, 0x2000000000, 0x4000000000, 0x8000000000,
-    0x10000000000, 0x20000000000, 0x40000000000, 0x80000000000,
-    0x100000000000, 0x200000000000, 0x400000000000, 0x800000000000,
-    0x1000000000000, 0x2000000000000, 0x4000000000000, 0x8000000000000,
-    0x10000000000000, 0x20000000000000, 0x40000000000000, 0x80000000000000,
-    0x100000000000000, 0x200000000000000, 0x400000000000000, 0x800000000000000,
-    0x1000000000000000, 0x2000000000000000, 0x4000000000000000, 0x8000000000000000
+    0x10000000, 0x20000000, 0x40000000, 0x80000000
 };
 
 static int number_of_bits(size_t x, long* result)
