diff --git a/python/mir/pyproject.toml b/python/mir/pyproject.toml
index fc2c37b..8df0ec5 100644
--- a/python/mir/pyproject.toml
+++ b/python/mir/pyproject.toml
@@ -9,7 +9,7 @@
 
 
 [build-system]
-requires = ["setuptools>=65", "wheel"]
+requires = ["setuptools>=65", "wheel", "Cython", "numpy"]
 build-backend = "setuptools.build_meta"
 
 [project]
diff --git a/python/mir/setup.py b/python/mir/setup.py
index 57b1bf3..3c4d726 100644
--- a/python/mir/setup.py
+++ b/python/mir/setup.py
@@ -1,7 +1,9 @@
 import platform
 import sys
 
-from setuptools import setup
+import numpy as np
+from Cython.Build import cythonize
+from setuptools import setup, Extension
 from wheel.bdist_wheel import bdist_wheel
 
 with open("VERSION", "r") as fVersion:
@@ -34,7 +36,18 @@ setup(
     package_data={
         "": ["*.so"],
     },
-    has_ext_modules=lambda: True,
+    ext_modules=cythonize([
+        Extension("mir._mir", [
+            "src/_mir/_mir.pyx",
+            "src/_mir/mir/appendix.cc",
+            "src/_mir/mir/input/ArrayInput.cc",
+            "src/_mir/mir/input/PyGribInput.cc",
+            "src/_mir/mir/output/PyGribOutput.cc",
+        ], language="c++", include_dirs=[
+            "/src/pyodide-recipes/packages/.libs/include",
+            np.get_include(),
+        ]),
+    ], compiler_directives={ "language_level": "3", "c_string_type": "unicode", "c_string_encoding": "utf8" }),
     install_requires=install_requires,
     **ext_kwargs[sys.platform],
 )
diff --git a/python/mir/src/_mir/_mir.pyx b/python/mir/src/_mir/_mir.pyx
index 8d6de6b..3d45994 100644
--- a/python/mir/src/_mir/_mir.pyx
+++ b/python/mir/src/_mir/_mir.pyx
@@ -11,6 +11,7 @@
 cimport eckit_defs as eckit
 cimport mir_defs as mir
 cimport numpy as cnp
+cnp.import_array()
 from cython.operator cimport dereference
 from libc.stdlib cimport free
 from libc.stdlib cimport malloc
