diff --git a/src/atlas/util/ConvexSphericalPolygon.h b/src/atlas/util/ConvexSphericalPolygon.h
index 985cc4e..4a80e62 100644
--- a/src/atlas/util/ConvexSphericalPolygon.h
+++ b/src/atlas/util/ConvexSphericalPolygon.h
@@ -153,7 +153,7 @@ private:
         size_t size() const { return size_; }
         size_t& size() { return size_; }
         double area() const;
-        std::array<SubTriangle, MAX_SIZE>::const_iterator end() const { return data() + size_; }
+        std::array<SubTriangle, MAX_SIZE>::const_iterator end() const { return begin() + size_; }
 
     private:
         size_t size_{0};
