diff --git a/setup.py b/setup.py
index f741c0a..cba088b 100644
--- a/setup.py
+++ b/setup.py
@@ -1073,8 +1073,9 @@ def _get_blosc_plugin():
     if "blosc" in get_system_clib_names():
         return HDF5PluginExtension(
             "hdf5plugin.plugins.libh5blosc",
-            sources=hdf5_blosc_sources,
-            include_dirs=get_clib_config("blosc", "include_dirs") + [hdf5_blosc_dir],
+            sources=hdf5_blosc_source + ["lib/hdf5_dl.c"],
+            include_dirs=get_clib_config("blosc", "include_dirs") + [hdf5_blosc_dir] + BuildConfig.get_hdf5_include_dirs(),
+            define_macros=[("H5_USE_18_API", None)],
             extra_compile_args=extra_compile_args,
             extra_link_args=get_clib_config("blosc", "extra_link_args")
             + extra_link_args,
@@ -1114,10 +1115,10 @@ def _get_blosc_plugin():
 
     return HDF5PluginExtension(
         "hdf5plugin.plugins.libh5blosc",
-        sources=sources + hdf5_blosc_sources,
+        sources=sources + hdf5_blosc_sources + ["lib/hdf5_dl.c"],
         extra_objects=get_clib_config("zstd", "extra_objects"),
-        include_dirs=include_dirs + [hdf5_blosc_dir],
-        define_macros=define_macros,
+        include_dirs=include_dirs + [hdf5_blosc_dir] + BuildConfig.get_hdf5_include_dirs(),
+        define_macros=define_macros + [("H5_USE_18_API", None)],
         extra_compile_args=extra_compile_args,
         extra_link_args=extra_link_args,
         libraries=libraries,
@@ -1151,8 +1152,9 @@ def _get_blosc2_plugin():
     if "blosc2" in get_system_clib_names():
         return HDF5PluginExtension(
             "hdf5plugin.plugins.libh5blosc2",
-            sources=hdf5_blosc2_sources,
-            include_dirs=get_clib_config("blosc2", "include_dirs") + [hdf5_blosc2_dir],
+            sources=hdf5_blosc2_sources + ["lib/hdf5_dl.c"],
+            include_dirs=get_clib_config("blosc2", "include_dirs") + [hdf5_blosc2_dir] + BuildConfig.get_hdf5_include_dirs(),
+            define_macros=[("H5_USE_18_API", None)],
             extra_compile_args=extra_compile_args,
             extra_link_args=get_clib_config("blosc2", "extra_link_args")
             + extra_link_args,
@@ -1220,10 +1222,10 @@ def _get_blosc2_plugin():
 
     return HDF5PluginExtension(
         "hdf5plugin.plugins.libh5blosc2",
-        sources=sources + hdf5_blosc2_sources,
+        sources=sources + hdf5_blosc2_sources + ["lib/hdf5_dl.c"],
         extra_objects=get_clib_config("zstd", "extra_objects"),
-        include_dirs=include_dirs + [hdf5_blosc2_dir],
-        define_macros=define_macros,
+        include_dirs=include_dirs + [hdf5_blosc2_dir] + BuildConfig.get_hdf5_include_dirs(),
+        define_macros=define_macros + [("H5_USE_18_API", None)],
         extra_compile_args=extra_compile_args,
         extra_link_args=extra_link_args,
         libraries=libraries,
@@ -1244,9 +1246,10 @@ def _get_zstandard_plugin():
 
     return HDF5PluginExtension(
         "hdf5plugin.plugins.libh5zstd",
-        sources=[f"{zstandard_dir}/zstd_h5plugin.c"],
+        sources=[f"{zstandard_dir}/zstd_h5plugin.c"] + ["lib/hdf5_dl.c"],
         extra_objects=get_clib_config("zstd", "extra_objects"),
-        include_dirs=[zstandard_dir] + get_clib_config("zstd", "include_dirs"),
+        include_dirs=[zstandard_dir] + get_clib_config("zstd", "include_dirs") + BuildConfig.get_hdf5_include_dirs(),
+        define_macros=[("H5_USE_18_API", None)],
         extra_link_args=get_clib_config("zstd", "extra_link_args"),
         libraries=get_clib_config("zstd", "libraries"),
     )
@@ -1281,12 +1284,13 @@ def _get_bitshuffle_plugin():
                 "bitshuffle_core.c",
                 "iochain.c",
             ],
-        ),
+        ) + ["lib/hdf5_dl.c"],
         extra_objects=get_clib_config("zstd", "extra_objects"),
         include_dirs=[bithsuffle_dir]
         + get_clib_config("lz4", "include_dirs")
-        + get_clib_config("zstd", "include_dirs"),
-        define_macros=define_macros,
+        + get_clib_config("zstd", "include_dirs")
+        + BuildConfig.get_hdf5_include_dirs(),
+        define_macros=define_macros + [("H5_USE_18_API", None)],
         extra_compile_args=extra_compile_args,
         extra_link_args=extra_link_args
         + get_clib_config("lz4", "extra_link_args")
@@ -1314,8 +1318,9 @@ def _get_lz4_plugin():
 
     return HDF5PluginExtension(
         "hdf5plugin.plugins.libh5lz4",
-        sources=["lib/LZ4/H5Zlz4.c", "lib/LZ4/lz4_h5plugin.c"],
-        include_dirs=get_clib_config("lz4", "include_dirs"),
+        sources=["lib/LZ4/H5Zlz4.c", "lib/LZ4/lz4_h5plugin.c"] + ["lib/hdf5_dl.c"],
+        include_dirs=get_clib_config("lz4", "include_dirs") + BuildConfig.get_hdf5_include_dirs(),
+        define_macros=[("H5_USE_18_API", None)],
         extra_compile_args=extra_compile_args,
         extra_link_args=get_clib_config("lz4", "extra_link_args"),
         libraries=libraries,
@@ -1329,9 +1334,9 @@ def _get_bzip2_plugin():
     """BZip2 plugin build config"""
     return HDF5PluginExtension(
         "hdf5plugin.plugins.libh5bzip2",
-        sources=["lib/PyTables/src/H5Zbzip2.c", "lib/H5Zbzip2_plugin.c"],
-        include_dirs=["lib/PyTables/src/"] + get_clib_config("bz2", "include_dirs"),
-        define_macros=[("HAVE_BZ2_LIB", 1)],
+        sources=["lib/PyTables/src/H5Zbzip2.c", "lib/H5Zbzip2_plugin.c"] + ["lib/hdf5_dl.c"],
+        include_dirs=["lib/PyTables/src/"] + get_clib_config("bz2", "include_dirs") + BuildConfig.get_hdf5_include_dirs(),
+        define_macros=[("HAVE_BZ2_LIB", 1)] + [("H5_USE_18_API", None)],
         extra_compile_args=[],
         extra_link_args=get_clib_config("bz2", "extra_link_args"),
         libraries=get_clib_config("bz2", "libraries"),
@@ -1361,13 +1366,13 @@ def _get_fcidecomp_plugin():
 
     return HDF5PluginExtension(
         "hdf5plugin.plugins.libh5fcidecomp",
-        sources=glob(f"{fcidecomp_dir}/fcicomp-*/src/*.c"),
-        include_dirs=glob(f"{fcidecomp_dir}/fcicomp-*/include") + charls_include_dirs,
+        sources=glob(f"{fcidecomp_dir}/fcicomp-*/src/*.c") + ["lib/hdf5_dl.c"],
+        include_dirs=glob(f"{fcidecomp_dir}/fcicomp-*/include") + charls_include_dirs + BuildConfig.get_hdf5_include_dirs(),
         extra_compile_args=extra_compile_args,
         extra_link_args=extra_link_args + get_clib_config("charls", "extra_link_args"),
         libraries=get_clib_config("charls", "libraries"),
         cpp14_required=True,
-        define_macros=macros,
+        define_macros=macros + [("H5_USE_18_API", None)],
     )
 
 
@@ -1383,8 +1388,9 @@ def _get_h5zfp_plugin():
 
     return HDF5PluginExtension(
         "hdf5plugin.plugins.libh5zfp",
-        sources=glob(f"{h5zfp_dir}/*.c"),
-        include_dirs=[f"{h5zfp_dir}/src"] + get_clib_config("zfp", "include_dirs"),
+        sources=glob(f"{h5zfp_dir}/*.c") + ["lib/hdf5_dl.c"],
+        include_dirs=[f"{h5zfp_dir}/src"] + get_clib_config("zfp", "include_dirs") + BuildConfig.get_hdf5_include_dirs(),
+        define_macros=[("H5_USE_18_API", None)],
         extra_compile_args=extra_compile_args,
         extra_link_args=get_clib_config("zfp", "extra_link_args"),
         libraries=get_clib_config("zfp", "libraries"),
@@ -1414,8 +1420,9 @@ def _get_sz_plugin():
 
     return HDF5PluginExtension(
         "hdf5plugin.plugins.libh5sz",
-        sources=glob(f"{h5zsz_dir}/src/*.c") + glob(f"{sz_dir}/src/*.c"),
-        include_dirs=include_dirs,
+        sources=glob(f"{h5zsz_dir}/src/*.c") + glob(f"{sz_dir}/src/*.c") + ["lib/hdf5_dl.c"],
+        include_dirs=include_dirs + BuildConfig.get_hdf5_include_dirs(),
+        define_macros=[("H5_USE_18_API", None)],
         extra_compile_args=extra_compile_args,
         extra_link_args=extra_link_args,
         libraries=get_clib_config("zlib", "libraries")
@@ -1449,9 +1456,10 @@ def _get_sz3_plugin():
 
     return HDF5PluginExtension(
         "hdf5plugin.plugins.libh5sz3",
-        sources=[f"{h5z_sz3_dir}/src/H5Z_SZ3.cpp"],
+        sources=[f"{h5z_sz3_dir}/src/H5Z_SZ3.cpp"] + ["lib/hdf5_dl.c"],
         extra_objects=get_clib_config("zstd", "extra_objects"),
-        include_dirs=include_dirs,
+        include_dirs=include_dirs + BuildConfig.get_hdf5_include_dirs(),
+        define_macros=[("H5_USE_18_API", None)],
         extra_compile_args=extra_compile_args,
         extra_link_args=extra_link_args + get_clib_config("zstd", "extra_link_args"),
         libraries=get_clib_config("zstd", "libraries"),
@@ -1474,12 +1482,12 @@ def _get_sperr_plugin():
                 "icecream.c",
                 "compactor.c",
             ],
-        ),
+        ) + ["lib/hdf5_dl.c"],
         include_dirs=get_clib_config("sperr", "include_dirs")
-        + [f"{h5z_sperr_dir}/include"],
+        + [f"{h5z_sperr_dir}/include"] + BuildConfig.get_hdf5_include_dirs(),
         extra_link_args=["-lstdc++"] + get_clib_config("sperr", "extra_link_args"),
         libraries=get_clib_config("sperr", "libraries"),
-        define_macros=get_clib_config("sperr", "macros"),
+        define_macros=get_clib_config("sperr", "macros") + [("H5_USE_18_API", None)],
         cpp20_required=True,
     )
 
@@ -1496,7 +1504,7 @@ _EMBEDDED_PLUGIN_EXTENSIONS = {
     "lz4": _get_lz4_plugin,
     "sperr": _get_sperr_plugin,
     "sz": _get_sz_plugin,
-    "sz3": _get_sz3_plugin,
+    # "sz3": _get_sz3_plugin,
     "zfp": _get_h5zfp_plugin,
     "zstd": _get_zstandard_plugin,
 }
@@ -1546,19 +1554,19 @@ def get_libraries_and_extensions():
         _EMBEDDED_PLUGIN_EXTENSIONS[name]() for name in embedded_extension_names
     ]
 
-    if extensions and sys.platform != "win32":
-        # Add hdf5 dynamic loading lib
-        libraries.append(
-            (
-                "hdf5_dl",
-                {
-                    "sources": ["lib/hdf5_dl.c"],
-                    "include_dirs": BuildConfig.get_hdf5_include_dirs(),
-                    "macros": [("H5_USE_18_API", None)],
-                    "cflags": [],
-                },
-            )
-        )
+    # if extensions and sys.platform != "win32":
+    #     # Add hdf5 dynamic loading lib
+    #     libraries.append(
+    #         (
+    #             "hdf5_dl",
+    #             {
+    #                 "sources": ["lib/hdf5_dl.c"],
+    #                 "include_dirs": BuildConfig.get_hdf5_include_dirs(),
+    #                 "macros": [("H5_USE_18_API", None)],
+    #                 "cflags": [],
+    #             },
+    #         )
+    #     )
 
     return libraries, extensions
 
