diff --git a/setup.py b/setup.py
index 20ecbcd..ce99701 100644
--- a/setup.py
+++ b/setup.py
@@ -1225,6 +1225,8 @@ def _get_blosc2_plugin():
     libraries += get_clib_config("zstd", "libraries")
     define_macros.append(("HAVE_ZSTD", 1))
 
+    define_macros.append(("BIT_STREAM_WORD_TYPE", "uint8"))
+
     return HDF5PluginExtension(
         "hdf5plugin.plugins.libh5blosc2",
         sources=sources + hdf5_blosc2_sources,
