diff --git a/lib/SZ/sz/src/callZlib.c b/lib/SZ/sz/src/callZlib.c
index 71d843d..b355bff 100644
--- a/lib/SZ/sz/src/callZlib.c
+++ b/lib/SZ/sz/src/callZlib.c
@@ -59,7 +59,7 @@ uint64_t zlib_compress(unsigned char* data, uint64_t dataLength, unsigned char**
 #endif
 
     uLong estCmpLen = deflateBound(&stream, dataLength);	
-	uint64_t outSize = estCmpLen;
+	uLong outSize = estCmpLen;
     	
 	*compressBytes = (unsigned char*)malloc(sizeof(unsigned char)*estCmpLen);
 	int err = compress2(*compressBytes, &outSize, data, dataLength, level);
@@ -265,7 +265,7 @@ uint64_t zlib_compress5(unsigned char* data, uint64_t dataLength, unsigned char*
 
 uint64_t zlib_uncompress(unsigned char* compressBytes, uint64_t cmpSize, unsigned char** oriData, uint64_t targetOriSize)
 {
-	uint64_t outSize = targetOriSize;
+	uLong outSize = targetOriSize;
 	*oriData = (unsigned char*)malloc(sizeof(unsigned char)*targetOriSize);	
 	int status = uncompress(*oriData, &outSize, compressBytes, cmpSize); 
 	if(status!=Z_OK)
diff --git a/lib/SZ/sz/src/dataCompression.c b/lib/SZ/sz/src/dataCompression.c
index 56db0c7..2326457 100644
--- a/lib/SZ/sz/src/dataCompression.c
+++ b/lib/SZ/sz/src/dataCompression.c
@@ -89,7 +89,7 @@ int64_t computeRangeSize_int(void* oriData, int dataType, size_t size, int64_t*
 	}
 	else if(dataType == SZ_INT64)
 	{
-		long* data = (int64_t *)oriData;
+		int64_t* data = (int64_t *)oriData;
 		int64_t data_;
 		min = data[0], max = min;
 		computeMinMax(data);
