diff --git a/python/pyarrow/__init__.py b/python/pyarrow/__init__.py
index 18a40d8..bb7fef9 100644
--- a/python/pyarrow/__init__.py
+++ b/python/pyarrow/__init__.py
@@ -29,6 +29,7 @@ streaming messaging and interprocess communication.
 For more information see the official page at https://arrow.apache.org
 """
 
+import unix_timezones
 import importlib as _importlib
 import os as _os
 import platform as _platform
