diff --git a/cfunits/units.py b/cfunits/units.py
index b1e4058..4f42ba8 100644
--- a/cfunits/units.py
+++ b/cfunits/units.py
@@ -1,6 +1,7 @@
 import ctypes
 import ctypes.util
 import operator
+from pathlib import Path
 
 from numpy import array as numpy_array
 from numpy import asanyarray as numpy_asanyarray
@@ -36,7 +37,7 @@ _ctypes_POINTER = {4: _POINTER(_c_float), 8: _POINTER(_c_double)}
 # else:
 #     # Linux
 #     _udunits = ctypes.CDLL('libudunits2.so.0')
-_libpath = ctypes.util.find_library("udunits2")
+_libpath = (Path(__file__) / ".." / ".." / "cfunits.libs" / "libudunits2.so").resolve()
 if _libpath is None:
     raise FileNotFoundError(
         "cfunits requires UNIDATA UDUNITS-2. Can't find the 'udunits2' "
